Fix Dart 2 issues in firebase plugins

mraleph opened the pull request:
Fixes https://github.com/flutter/flutter/issues/15027

mraleph commented:
/fyi @mravn-google I have fixed all `firebase_*` plugins except `firebase_database` where test crashes and I don't have time for it right now.

tvolkert commented:
Looks like Travis isn't happy:

```
Checking project firebase_auth...
  error - Missing parameter type for 'v' at lib/firebase_auth.dart:111:71 - (strong_mode_implicit_dynamic_parameter)
  info - Specify type annotations at lib/firebase_auth.dart:111:71 - (always_specify_types)
  info - Avoid using `as` at lib/firebase_auth.dart:111:77 - (avoid_as)
```

mravn-google commented on packages/firebase_auth/lib/firebase_auth.dart:
```diff
@@ -108,7 +108,11 @@ class FirebaseAuth {
     StreamController<FirebaseUser> controller;
     controller = new StreamController<FirebaseUser>.broadcast(onListen: () {
-      _handle = channel.invokeMethod('startListeningAuthState').then((v) => v as int);
+      _handle =
```
Could we do
```dart
_handle = channel.invokeMethod('startListeningAuthState').then<int>((dynamic v) => v);
```

mraleph replied:
Done.

mraleph commented:
I have fixed more issues. Where I fixed the plugin code itself I bumped version in pubspec and added an entry to CHANGELOG. 

Now all the tests except one test in firebase_database pass. The test in firebase_database fails due to https://github.com/dart-lang/sdk/issues/28498. 

@mravn-google please take another quick look

mraleph commented:
@tvolkert Travis seems to be stuck on one of the Shards - do you know if there is a way to nudge it?

mravn-google commented:
I restarted the job.

mraleph commented:
Now, I am getting errors in the files I did not edit :-/ I am just gonna push it as is.

tvolkert commented:
It looks like `dartfmt` has been updated with some improvements, but it's yielding an error because our existing formatted code isn't formatted correctly anymore.

cdock1029 commented:
When will this be available through pub? Not seeing 0.4.7 yet

https://pub.dartlang.org/packages/firebase_auth#-versions-tab-

mravn-google commented:
Sorry, that was an omission on my part. Version 0.4.7 has just been published.
